### PR TITLE
generate video images

### PR DESCRIPTION
**Pull request by muhammad-ammar**
[EDUCATOR-205](https://openedx.atlassian.net/browse/EDUCATOR-205)

### CODE REVIEW

File: default_config.yaml [hunk 1]
```diff
@@ -21,6 +26,7 @@ val_password:
 val_secret_key: 
 val_token_url: 
 val_username: 
+val_video_images_url:
```
Comment (muhammad-ammar) [inline default_config.yaml]: We will create a new view for this in edxval.

File: default_config.yaml [hunk 2]
```diff
@@ -13,6 +13,11 @@ aws_storage_bucket:
 multi_upload_barrier: 2000000000
 boto_timeout: 60
 
+## AWS Storage config for video images
+aws_video_images_access_key:
+aws_video_images_secret_key:
+aws_video_images_bucket:
+aws_video_images_prefix:
```
Comment (muhammad-ammar) [inline default_config.yaml]: @yro Are these things enough to upload images to S3 bucket?

**Comment by muhammad-ammar** [comment]
@yro These are changes i have made so far. This is not yet completed because we need testing and also a new DRF view is required on edxval which will be used by VEDA to store auto generated images bucket keys.

Can you review these changes? Also what is the best way you think to test these changes to see if they really work?

File: default_config.yaml (hunk 2 above)
Comment (yro) [inline default_config.yaml]: @muhammad-ammar Yep - should be fine.

File: veda_worker/video_images.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                message='Video Images generation failed: No Video Object'
+            )
+            return None
+
+        # We need to generate images from different positions
+        # starting from 5 second. We choose 5 to skip initial
+        # credits shown in video.
+        step = self.video_object.mezz_duration/IMAGE_COUNT
+        positions = [5 + i * step for i in range(IMAGE_COUNT)]
+        generated_images = []
```
Comment (yro) [inline veda_worker/video_images.py]: Nice. Smart.

**Comment by yro** [comment]
@muhammad-ammar There's no strong way of testing, unfortunately. I can run this in a sandboxed environment with credentials to make sure everything is working on the VEDA end. Once the VAL view is implemented we can run a better end-to-end test.

**Comment by muhammad-ammar** [comment]
@yro [VAL view](https://github.com/edx/edx-val/pull/74) to update auto generated images is ready.

**Comment by yro** [comment]
@muhammad-ammar Great! I'll test this out today and get it merged. Is this okay to deploy and start generating images?

**Comment by yro** [comment]
👍  Very nice, thank you @muhammad-ammar. It's an ugly repo and you did really good work. I'll merge and consult with devops on deployment.

File: veda_worker/__init__.py [hunk 4]
```diff
@@ -154,14 +154,15 @@ def run(self):
...
         # generate video images command and update S3 and edxval
-        VideoImages(
-            video_object=self.VideoObject,
-            work_dir=self.workdir,
-            source_file=self.source_file,
-            jobid=self.jobid
-        ).create_and_update()
-
+        # run against 'hls' encode only
         if self.encode_profile == 'hls':
```
Comment (muhammad-ammar) [inline veda_worker/__init__.py]: But we need to generate images always. Why only in case of hls?

Comment (yro) [inline veda_worker/__init__.py]: The different encode streams run in parallel, but we only need to generate the images once per video, so it makes sense to tie them to a single encode. I am kind of against hard coding it like this, but it's the encode stream we know we'll always be making.

**Pull request merged**